### PR TITLE
[Utility Renaming] Move the remaining headers from core/char_oprations to utility/char_operations

### PR DESCRIPTION
**Pull request by smehringer**
Part of https://github.com/seqan/product_backlog/issues/160

### CODE REVIEW

File: include/seqan3/utility/char_operations/predicate_detail.hpp [hunk 1]
```diff
@@ -15,10 +15,11 @@
 
 #include <array>
 #include <cctype>
-#include <seqan3/std/concepts>
 #include <stdexcept>
 #include <string>
 
+#include <seqan3/std/concepts>
+
```
Comment (joergi-w) [inline include/seqan3/utility/char_operations/predicate_detail.hpp:22]: Why did you move this down here? I understood that we sort it as if it was `<concepts>`...

File: include/seqan3/utility/char_operations/all.hpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #pragma once
 
 #include <seqan3/utility/char_operations/predicate.hpp>
+#include <seqan3/utility/char_operations/pritty_print.hpp>
```
Comment (joergi-w) [inline include/seqan3/utility/char_operations/all.hpp:17]: Here is a typo: => pretty

File: include/seqan3/utility/char_operations/pretty_print.hpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides pretty printing for chars.
```
Comment (joergi-w) [inline include/seqan3/utility/char_operations/pretty_print.hpp:10]: I know that this is not the scope of this PR, however users may not know what we consider as "pretty printing". Can we find a higher level description, e.g. "Provides the transformation of chars to a visual representation/ to a printable string" or similar?

File: include/seqan3/core/char_operations/all.hpp [hunk 4]
```diff
@@ -6,16 +6,21 @@
...
 #include <seqan3/utility/char_operations/all.hpp>
 
+SEQAN3_DEPRECATED_HEADER(
+   "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/utility/char_operations/all.hpp> instead.")
+
 /*!\defgroup char_operations builtin_character operations
- * \brief Provides various operations on character types.
+ * \brief [DEPRECATED] Provides various operations on character types.
  * \ingroup core
+ * \deprecated This submodule will be removed in 3.1. Please see the utility/char_operations submodule instead.
```
Comment (joergi-w) [inline include/seqan3/core/char_operations/all.hpp:25]: A doxygen group can just be deleted here and newly created in the utility module I think. Otherwise, you may have name conflicts.

**Review by joergi-w — CHANGES_REQUESTED** [review]

File: include/seqan3/utility/char_operations/all.hpp (hunk 2 above)
Comment (smehringer) [reply include/seqan3/utility/char_operations/all.hpp:17]: good catch

Comment (smehringer) [reply include/seqan3/utility/char_operations/all.hpp:17]:
```suggestion
#include <seqan3/utility/char_operations/pretty_print.hpp>
```

File: include/seqan3/utility/char_operations/predicate_detail.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/utility/char_operations/predicate_detail.hpp:22]: whoops. good catch. I don't know what happend

File: include/seqan3/utility/char_operations/pretty_print.hpp (hunk 3 above)
Comment (smehringer) [reply include/seqan3/utility/char_operations/pretty_print.hpp:10]: How about:

```suggestion
 * \brief Provides seqan3::detail::make_printable which converts non printable characters (e.g. a tab) to a std::string.
```

Comment (joergi-w) [reply include/seqan3/utility/char_operations/pretty_print.hpp:10]: Sounds great!

**Review by joergi-w — APPROVED** [review]

File: include/seqan3/core/char_operations/all.hpp [hunk 5]
```diff
@@ -6,18 +6,15 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides Char predicates for tokenization.
+ * \brief [DEPRECATED] Provides Char predicates for tokenization.
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \deprecated This header will be removed in 3.1. Please use seqan3/utility/char_operations/all.hpp instead.
```
Comment (rrahn) [inline include/seqan3/core/char_operations/all.hpp:12]: The message is not as given on the original ticket. Please update the message.

File: include/seqan3/core/char_operations/predicate_detail.hpp [hunk 6]
```diff
@@ -6,382 +6,15 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides parse conditions for tokenization.
+ * \brief [DEPRECATED] Provides parse conditions for tokenization.
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \deprecated This header will be removed in 3.1. Please use seqan3/utility/char_operations/predicate_detail.hpp instead.
```
Comment (rrahn) [inline include/seqan3/core/char_operations/predicate_detail.hpp:12]: same here

File: include/seqan3/core/char_operations/pretty_print.hpp [hunk 7]
```diff
@@ -7,64 +7,13 @@
 
 /*!\file
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
- * \brief Provides pretty printing for chars.
+ * \brief [DEPRECATED] Provides pretty printing for chars.
+ * \deprecated This header will be removed in 3.1. Please use seqan3/utility/char_operations/pretty_print.hpp instead.
```
Comment (rrahn) [inline include/seqan3/core/char_operations/pretty_print.hpp:11]: same here

File: include/seqan3/core/char_operations/transform.hpp [hunk 8]
```diff
@@ -6,99 +6,14 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides utilities for modifying characters.
+ * \brief [DEPRECATED] Provides utilities for modifying characters.
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \deprecated This header will be removed in 3.1. Please use seqan3/utility/char_operations/transform.hpp instead.
```
Comment (rrahn) [inline include/seqan3/core/char_operations/transform.hpp:11]: same here

**Review by rrahn — CHANGES_REQUESTED** [review]
Looks good. Just the doxygen deprecation messages seem to deviate from the proposed text of the ticket.

**Review by rrahn — APPROVED** [review]
thank you
